packages/runner/src/scanner/page.ts

```diff
@@ -27,28 +28,39 @@ export class Page {
         return this.puppeteerPage.setBypassCSP(true);
     }
 
-    public async goto(url: string): Promise<void> {
+    public async scanForA11yIssues(url: string): Promise<AxeScanResults> {
         const gotoUrlPromise = this.puppeteerPage.goto(url, { waitUntil: ['load'] });
         const waitForNetworkLoadPromise = this.puppeteerPage.waitForNavigation({ waitUntil: ['networkidle0'], timeout: 15000 });
 
-        await gotoUrlPromise;
+        const response = await gotoUrlPromise;
+
+        if (!this.isHtmlPage(response)) {
+            return { unscannable: true, error: `Cannot scan ${url} because it is not a html page.` };
+        }
 
         try {
             // We ignore error if the page still has network activity after 15 sec
             await waitForNetworkLoadPromise;
             // tslint:disable-next-line:no-empty
         } catch {}
-    }
 
-    public async scanForA11yIssues(): Promise<AxeResults> {
         const axePuppeteer: AxePuppeteer = this.axePuppeteerFactory(this.puppeteerPage);
-
-        return axePuppeteer.analyze();
+        const scanResults = await axePuppeteer.analyze();
+        return { results: scanResults };
     }
 
     public async close(): Promise<void> {
         if (this.puppeteerPage !== undefined) {
             await this.puppeteerPage.close();
         }
     }
+
+    private isHtmlPage(response: Puppeteer.Response): boolean {
+        const contentType = this.getContentType(response.headers());
+        return contentType != null && contentType.indexOf('text/html') != -1;
+    }
+
+    private getContentType(headers: Record<string, string>): string {
+        return headers['content-type'] || headers['Content-Type'];
```

Comment:
>headers['content-type'] || headers['Content-Type']; [](start = 15, length = 51)

or something like this:
Object.keys(headers).filter(p => p.toLowercase() == content-type).first();

Reply:
Discussed offline, no need to check the string case here since they mentioned ``All header names are lower-case`` in [puppeteer API doc](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#responseheaders)